inout.c

```diff
@@ -203,3 +289,25 @@ uint8out(PG_FUNCTION_ARGS)
 	sprintf(result, "%"PRIu64, (uint64_t) arg1);
 	PG_RETURN_CSTRING(result);
 }
+
+PG_FUNCTION_INFO_V1(uint8recv);
+Datum
+uint8recv(PG_FUNCTION_ARGS)
+{
+	StringInfo buf = (StringInfo) PG_GETARG_POINTER(0);
+
+	PG_RETURN_UINT64((uint64) pq_getmsgint(buf, sizeof(uint64)));
```

Comment:
It looks like there's a bug here. `pq_getmsgint` errors with `unsupported integer size 8`. https://github.com/postgres/postgres/blob/c30f54ad732ca5c8762bb68bbe0f51de9137dd72/src/backend/libpq/pqformat.c#L439

Looks like the right function is `pq_getmsgint64`
https://github.com/postgres/postgres/blob/c30f54ad732ca5c8762bb68bbe0f51de9137dd72/src/backend/libpq/pqformat.c#L455

Reply:
You are correct. pq_getmsgint64 should be used there